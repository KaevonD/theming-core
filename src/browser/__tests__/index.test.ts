// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import { preset, override } from '../../__fixtures__/common';
import { applyTheme } from '../index';

afterEach(() => {
  allStyleNodes().forEach((tag) => tag.remove());
});

test('attaches one style node containing override', () => {
  applyTheme({ override, preset });

  const styleNodes = allStyleNodes();

  expect(styleNodes).toHaveLength(1);
  const themeNode = styleNodes[0];

  expect(themeNode.innerHTML).toMatchSnapshot();
});

test('removes style node on reset', () => {
  const { reset } = applyTheme({ override, preset });

  reset();

  expect(allStyleNodes()).toHaveLength(0);
});

const allStyleNodes = () => document.head.querySelectorAll('style');
