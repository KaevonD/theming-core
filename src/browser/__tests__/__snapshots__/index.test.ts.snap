// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attaches one style node containing override 1`] = `
":root:root{
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:yellow;
	--buttonShadow-css:red;
	--boxShadow-css:green;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:3px;
	--appear-css:20ms;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.compact.compact:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:1px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.dark.dark:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:orange;
	--buttonShadow-css:red;
	--boxShadow-css:brown;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.disabled-motion.disabled-motion:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--appear-css:0;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.navigation.navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:pink;
	--buttonShadow-css:red;
	--boxShadow-css:purple;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:3px;
	--appear-css:20ms;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.compact.compact .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:1px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.compact.compact.navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:1px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.dark.dark .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:brown;
	--buttonShadow-css:green;
	--boxShadow-css:purple;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.dark.dark.navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:brown;
	--buttonShadow-css:green;
	--boxShadow-css:purple;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.disabled-motion.disabled-motion .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--appear-css:0;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.disabled-motion.disabled-motion.navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--appear-css:0;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}"
`;
